scripts/CEE/drain-reboot-nodes/script.sh

```diff
@@ -0,0 +1,115 @@
+#!/bin/bash
+#
+# Drain and Reboot a worker node.
+# Input:
+#  (Env) WORKER
+#    name of the worker node to be drained / rebooted
+
+set -e
+set -o nounset
+set -o pipefail
+
+CURRENTDATE=$(date +"%Y-%m-%d %T")
+
+if [[ -z "${WORKER}" ]]; then
+    echo "[Error] Variable WORKER cannot be blank."
+    exit 1
+fi
+
+start_job(){
+    echo "Job started at $CURRENTDATE"
+    echo ".................................."
+    echo
+}
+
+finish_job(){
+    echo
+    echo ".................................."
+    echo "Job finished at $CURRENTDATE"
+}
+
+## Function which checks if the node is a Worker node
+check_worker(){
+
+    echo "Checking if \"${WORKER}\" is a worker node..."
+    
+    if (oc get nodes -l node-role.kubernetes.io/worker=,node-role.kubernetes.io/infra!= -oname | grep "${WORKER}") &> /dev/null; then
+        echo "[OK] \"${WORKER}\" is a worker node. Proceeding with draining..."
+        echo
+    else
+        echo "[Error] \"${WORKER}\" is not a worker node. Exiting script..."
+        exit 1
+    fi
+
+}
+
+## Function which drains the node bypassing the PDB check
+drain_worker(){
+
+    echo "Draining node \"${WORKER}\"..."
+    oc adm drain "${WORKER}" --ignore-daemonsets --delete-emptydir-data --force --disable-eviction
```

Comment:
I overlooked this when suggesting the RBAC changes.  I suppose this command still need get/list/delete pod permission cluster-wide?  I haven't tested it though.   Could you help to test it when possible?

Reply:
Done.